lua/luasnip/session/snippet_collection.lua

```diff
@@ -128,6 +130,33 @@ local by_id = setmetatable({}, {
 	__mode = "v",
 })
 
+-- If `add_snippets` opts.source and config.store_meta_data then each call to
+-- add snippets stores the source file and potentially other meta data in this
+-- table which then each snippet points to so that one can `jump-to-snip`
+-- by calling `loaders.edit_snippet_files({ target_snippet = snip_context_x })`
+local snippets_meta_data = {}
+local snip_id_to_meta_map = {}
+
+local function initialize_snippet_meta_data(opts)
```

Comment:
Okay, I think this mechanism is good, but we need to split it up a bit:
* `config` generates a function which, given a filename produces the metadata that shall be added to snippets loaded by one of the loaders. We should not pass the filename directly, but wrap it in a table like `{filename=<the filename>}`, since we can then add more arguments and the function will stay comfortable to use (no need to write a handler like `function(_, _, _, theOneArgINeed) ... end`)
* this function is used in loaders to generate the metadata for the snippets (no ifs or something like that, we can just call the function. This is flexible enough to probably not require any changes, except if new parameters are added)
* `add_snippets` receives `metadata`-table, which is associated with the snippets added in this call, unconditionally

Reply:
Ok, I have to ponder your reply and come back later with a revised version.

Reply:
Looking forward to it :D
If you have any questions or other ideas, go ahead, I'd love to find something better (especially the association of snippet with its metadata is a bit annoying, since we have to do that id-redirection. Can't think of a better way though :/)

Reply:
Cool, yeah i cant really come up with something better either atm.